### PR TITLE
Skip texture management example on MI300 and above

### PR DESCRIPTION
**Pull request by dgaliffiAMD**
To resolve build failure when building the entire example suite on an MI300.

### CODE REVIEW

File: HIP-Basic/CMakeLists.txt [hunk 1]
```diff
@@ -127,7 +127,19 @@ if(NOT WIN32 AND NOT "${GPU_RUNTIME}" STREQUAL "HIP")
...
+
+# This example is not supported on gfx940, gfx941, and gfx942
+# See: __HIP_NO_IMAGE_SUPPORT in hip/amd_detail/amd_device_functions.h
+list(REMOVE_DUPLICATES CMAKE_HIP_ARCHITECTURES)
+if(NOT (CMAKE_HIP_ARCHITECTURES MATCHES "gfx94*"))
+    add_subdirectory(texture_management)
+else()
+    message(
+        WARNING
+        "SKIP: Texture management example not supported on gfx940, gfx941, and gfx942"
```
Comment (Beanavil) [inline HIP-Basic/CMakeLists.txt]: Nit: I would write it more similar to the other messages we print when skipping examples. Something like:
```suggestion
        "Texture functions not supported on gfx94*, not building texture management example"
```

File: HIP-Basic/texture_management/CMakeLists.txt [hunk 2]
```diff
@@ -57,23 +57,36 @@ endif()
...
+message(
+    STATUS
+    "texture_management - CMAKE_HIP_ARCHITECTURES: ${CMAKE_HIP_ARCHITECTURES}"
 )
 
-set(include_dirs "../../Common")
-if(GPU_RUNTIME STREQUAL "CUDA")
-    list(APPEND include_dirs "${ROCM_ROOT}/include")
-endif()
+if(NOT CMAKE_HIP_ARCHITECTURES MATCHES "gfx94*")
```
Comment (Beanavil) [inline HIP-Basic/texture_management/CMakeLists.txt:67]: If we don't add the folder on `HIP-Basic/CMakeLists.txt` this change does not seem necessary IINM. I would suggest either keeping this and removing the skipping code on `HIP-Basic/CMakeLists.txt` or vice-versa.

**Comment by Beanavil** [comment]
FYI: in our internal PR for fixing the examples for ROCm 6.1 we also solved this issue by doing
```diff
+ #if !defined(__HIP_NO_IMAGE_SUPPORT) || !__HIP_NO_IMAGE_SUPPORT
+   // Read the value from the texture, if supported.
     unsigned char val = tex2D<unsigned char>(tex_obj, u, v);
+ #else
+   // Prevent compile errors on HIP architectures that do not support
+    // texture instructions.
+    (void)u;
+   (void)v;
+   (void)tex_obj;
+   unsigned char val = 0;
+ #endif
```
within the texture management example, so it can still be compiled for the gfx94* archs.

**Comment by dgaliffiAMD** [comment]
> also solved this issue by doing



> FYI: in our internal PR for fixing the examples for ROCm 6.1 we also solved this issue by doing
> 
> ```diff
> + #if !defined(__HIP_NO_IMAGE_SUPPORT) || !__HIP_NO_IMAGE_SUPPORT
> +   // Read the value from the texture, if supported.
>      unsigned char val = tex2D<unsigned char>(tex_obj, u, v);
> + #else
> +   // Prevent compile errors on HIP architectures that do not support
> +    // texture instructions.
> +    (void)u;
> +   (void)v;
> +   (void)tex_obj;
> +   unsigned char val = 0;
> + #endif
> ```
> 
> within the texture management example, so it can still be compiled for the gfx94* archs.

Sorry for the late reply. I'm just getting back from vacation. 
Thanks for the FYI. Will your internal change eventually be pushed to ROCm/rocm-example? 
If yes, I'll just wait for that to come. If not, I'll update my PR. :)

**Comment by Beanavil** [comment]
> Thanks for the FYI. Will your internal change eventually be pushed to ROCm/rocm-example?
If yes, I'll just wait for that to come. If not, I'll update my PR. :)

@dgaliffiAMD yes, just opened the PR now (#138 ) :)

**Comment by dgaliffiAMD** [comment]
Closing, as this will be addressed in PR#138.

**Pull request closed without merging**